tests/compilers/native_functions.rs

```diff
@@ -77,34 +76,34 @@ fn native_function_works_for_wasm() -> Result<()> {
     Ok(())
 }
 
-#[test]
-#[should_panic(
-    expected = "Closures (functions with captured environments) are currently unsupported with native functions. See: https://github.com/wasmerio/wasmer/issues/1840"
-)]
-fn native_host_function_closure_panics() {
-    let store = get_store(false);
-    let state = 3;
-    Function::new_native(&store, move |_: i32| {
-        println!("{}", state);
-    });
-}
-
-#[test]
-#[should_panic(
-    expected = "Closures (functions with captured environments) are currently unsupported with native functions. See: https://github.com/wasmerio/wasmer/issues/1840"
-)]
-fn native_with_env_host_function_closure_panics() {
-    let store = get_store(false);
-    let state = 3;
-    let env = 4;
-    Function::new_native_with_env(&store, env, move |_env: &i32, _: i32| {
-        println!("{}", state);
-    });
-}
-
-#[test]
-fn non_native_functions_and_closures_with_no_env_work() -> Result<()> {
-    let store = get_store(false);
+// #[should_panic(
+//     expected = "Closures (functions with captured environments) are currently unsupported with native functions. See: https://github.com/wasmerio/wasmer/issues/1840"
+// )]
+// #[compiler_test(native_functions)]
+// fn native_host_function_closure_panics(config: crate::Config) {
+//     let store = config.store();
+//     let state = 3;
+//     Function::new_native(&store, move |_: i32| {
+//         println!("{}", state);
+//     });
+// }
+
+// #[should_panic(
+//     expected = "Closures (functions with captured environments) are currently unsupported with native functions. See: https://github.com/wasmerio/wasmer/issues/1840"
+// )]
+// #[compiler_test(native_functions)]
+// fn native_with_env_host_function_closure_panics(config: crate::Config) {
+//     let store = config.store();
+//     let state = 3;
+//     let env = 4;
+//     Function::new_native_with_env(&store, env, move |_env: &i32, _: i32| {
+//         println!("{}", state);
+//     });
+// }
```

Comment:
Why are they commented now?

Reply:
It was because the `compiler_test` macro didn't support properly the function decorators.
I just added support and put the code as it was!